File: NSFlights_Frontend/src/app/model/Journey.ts
export class Journey {
    origin: string;
    destination: string;
    price: number;
    flights: Flight[];

    constructor(
        origin: string, destination: string, price: number, flights: Flight[]
    ) {
        this.origin = origin;
        this.destination = destination;
        this.price = price;
        this.flights = flights;
    }
}

export class Flight {
    origin: string;
    destination: string;
    transport: Transport;
    price: number

    constructor(
        origin: string, destination: string, transport: Transport, price: number,
    ) {
        this.origin = origin;
        this.destination = destination;
        this.transport = transport;
        this.price = price;
    }
}

export class Transport {
    flightCarrier: string;
    flightNumber: string;

    constructor(flightCarrier: string, flightNumber: string) {
        this.flightCarrier = flightCarrier
        this.flightNumber = flightNumber
    }
}

File: NSFlights_Backend/NSFlights.Model/Journey.cs
﻿
namespace Sales.Model
{
    public class Journey
    {
        public string DepartureStation { get; set; }
        public string ArrivalStation { get; set; }
        public string FlightCarrier { get; set; }
        public string FlightNumber { get; set; }
        public double Price { get; set; }

        public Journey(string departureStation, string arrivalStation, string flightCarrier, string flightNumber, double price)
        {
            DepartureStation = departureStation;
            ArrivalStation = arrivalStation;
            FlightCarrier = flightCarrier;
            FlightNumber = flightNumber;
            Price = price;
        }
    }
}
